File: April 4, 2026.js
function isValidEquation(equation) {

  const [leftSide, rightSide] = equation.split("=");
  const tokens = leftSide.match(/\d+|[+\-*/]/g);
  const rightN = Number(rightSide.trim())

  for (let i = 0; i < tokens.length; i++) {
    if (tokens[i] == "*" || tokens[i] == "/") {
      let prevN = Number(tokens[i - 1]);
      let nextN = Number(tokens[i + 1]);
      let parcialResult = 0;
      if (tokens[i] == "*") {
        parcialResult = prevN * nextN;
      } else {
        parcialResult = prevN / nextN;
      }

      tokens.splice(i - 1, 3, parcialResult.toString());
      i--;
    }
  }
  
  let result = Number(tokens[0]);

  for (let i = 1; i < tokens.length; i += 2) {
    let op = tokens[i];
    let n = Number(tokens[i + 1]);

    if (op == "+") {
        result += n;
      } else {
        result -= n;
      }
  }

  return result == rightN;
}

File: April 4, 2026.py
import re

def is_valid_equation(equation):

    left_side, right_side = equation.split("=")
    tokens = re.findall(r'\d+|[+\-*/]', left_side)
    right_n = float(right_side.strip())

    i = 0
    for _ in range(len(tokens)):
        
        if i >= len(tokens):
            break

        if tokens[i] in ("*", "/"):
            prev_n = float(tokens[i - 1])
            next_n = float(tokens[i + 1])

            if tokens[i] == "*":
                parcial_result = prev_n * next_n
            else:
                parcial_result = prev_n / next_n

            tokens[i - 1:i + 2] = [str(parcial_result)]
            i -= 1
        else:
            i += 1

    result = float(tokens[0])

    i = 1
    for _ in range(len(tokens) // 2):
        
        if i >= len(tokens):
            break

        op = tokens[i]
        n = float(tokens[i + 1])

        if op == "+":
            result += n
        else:
            result -= n

        i += 2

    return result == right_n